Realm/ObjectServerTests/SwiftCollectionSyncTests.swift

```diff
@@ -0,0 +1,391 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+import RealmSwift
+import XCTest
+
+#if canImport(RealmTestSupport)
+import RealmSwiftSyncTestSupport
+import RealmSyncTestSupport
+import RealmTestSupport
+#endif
+
+class ListSyncTests: SwiftSyncTestCase {
+    private func roundTrip<T: _ManagedPropertyType>(keyPath: KeyPath<SwiftCollectionSyncObject, List<T>>,
+                                                    values: [T],
+                                                    partitionValue: String = #function) throws {
+        let user = logInUser(for: basicCredentials(withName: partitionValue,
+                                                   register: isParent))
+        let realm = try openRealm(partitionValue: partitionValue, user: user)
+        if isParent {
+            checkCount(expected: 0, realm, SwiftCollectionSyncObject.self)
+            executeChild()
+            waitForDownloads(for: realm)
+            checkCount(expected: 1, realm, SwiftCollectionSyncObject.self)
+            // Run the child again to add the values
+            executeChild()
+            waitForDownloads(for: realm)
+            checkCount(expected: 1, realm, SwiftCollectionSyncObject.self)
+            let object = realm.objects(SwiftCollectionSyncObject.self).first!
+            XCTAssertEqual(object[keyPath: keyPath].count, values.count*2)
```

Comment:
A lot of this code would be more readable if `object[keyPath: keyPath]` was stored in a local variable.